packages/main/src/StepInput.ts

```diff
@@ -366,8 +375,16 @@ class StepInput extends UI5Element implements IFormElement {
 		return this.focused;
 	}
 
-	get _valuePrecisioned() {
-		return this.value.toFixed(this.valuePrecision);
+	get _displayValue() {
+		if ((this.value === 0 && this._initialRender) || (Number.isInteger(this.value) && !this.valuePrecision)) {
+			return this.value.toFixed(this.valuePrecision);
+		}
+
+		if (this.value === Number(this.input.value)) { // For the cases where the number is fractional and is ending with 0s.
```

Comment:
This is the same as '==' . But I pref this way, because is more descriptive and it's not overcomplicating the code, because JS will do the casting anyway.